styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -244,6 +253,7 @@ public static StyxScheduler createDefault() {
   private final RetryUtil retryUtil;
   private final WorkflowResourceDecorator resourceDecorator;
   private final EventConsumerFactory eventConsumerFactory;
+  private final WorkflowConsumerFactory workflowConsumerFactory;
```

Comment:
Not a big fan of `consumer` naming here, since it doesn't really consume workflow.

Reply:
This was chosen also to be consistent with `EventConsumerFactory`. Any suggestion for a better name?

Reply:
Not sure. This consumes the state change, or update, not the workflow itself. I really have no good suggestion. So it is ok.